src/components/LayoutTree.tsx

```diff
@@ -414,6 +417,8 @@ const TreeNode = ({
 } & Pick<CSSTransitionProps, 'in'>) => {
   const [y, setY] = useState(_y)
   const fadeThoughtRef = useRef<HTMLDivElement>(null)
+  const lastPatches = useSelector(state => state.undoPatches[state.undoPatches.length - 1])
+  const isLastActionNewThought = lastPatches?.some(patch => patch.actions[0] === 'newThought')
```

Comment:
I recommend combining `lastPatches` and `isLastActionNewThought` into a single `useSelector` for performance reasons. Then the render function will only be re-triggered when `isLastActionNewThought` changes, instead of every time `lastPatches` changes.

In general aim for "thick" selectors in Redux for better performance. In many cases it won't be relevant, but we know from testing that `LayoutTree` is a heavy component and takes several milliseconds to render.

Reply:
`isLastActionNewThought` is now updated in a single `useSelector`.